pkg/configprovider/providerutil/utils.go

```diff
@@ -0,0 +1,45 @@
+package providerutil
+
+import (
+	"go.opentelemetry.io/collector/confmap"
+	"strings"
+)
+
+func removeKeyFromSrcMap(srcMap map[string]interface{}, mergeMap map[string]interface{}, keys []string, index int) map[string]interface{} {
+
+	if index == len(keys) { //out of index
+		return srcMap
+	}
+
+	currentKey := keys[index]
+	leafIndex := len(keys) - 1
+	if index == leafIndex { // Got Leaf key, if leaf key exists in both maps, remove it from source map
+		_, existInSrc := srcMap[currentKey]
+		_, existInMerge := mergeMap[currentKey]
+		// If leaf key exists in both maps, remove from source map
+		if existInSrc && existInMerge {
+			delete(srcMap, currentKey)
+		}
```

Comment:
We can `return` here and then we can remove `else` statement (unnecessary nesting lowers readability)